### PR TITLE
release-notes.txt: correct some mistakes

### PR DESCRIPTION
**Pull request by kYc0o**
I found very little mistakes on the release notes regarding issue numbers and descriptions.

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -209,7 +208,6 @@ other issues
 #3366: periph/i2c: handle NACK
 #4488: Making the newlib thread-safe: When calling puts/printf after thread_create(), the CPU hangs
        for DMA enabled uart drivers.
-#4866: periph: GPIO drivers are not thread safe
```
Comment (miri64) [inline release-notes.txt]: This is still open AFAIK...

Comment (kYc0o) [reply release-notes.txt]: It's a duplicate, see above.

Comment (miri64) [reply release-notes.txt]: You mean [L185](https://github.com/RIOT-OS/RIOT/pull/7917/files#diff-acce5952fb2ed7b16765d3ddab1e2403L186)? Alright then.

Comment (kYc0o) [reply release-notes.txt]: Yes, though the description doesn't match only with the title of the issue but also with the description:

```
#4866: not all GPIO driver implementations are thread safe: Due to non-atomic operations in the drivers
```

**Comment by miri64** [comment]
Please provide a backport to the 2017.10 release branch.

**Comment by kYc0o** [comment]
Backport in #7924

**Pull request merged**